app/representers/concerns/embeds.rb

```diff
@@ -56,10 +56,23 @@ module ClassMethods
 
     def embed(name, options={})
       options[:embedded] = true
+      options[:writeable] = false
+      options[:if] ||= ->(_a){ self.id } unless options[:zoom] == :always
 
       if options[:paged]
-        opts = {no_curies: true, item_class: options.delete(:item_class), url: options.delete(:url), item_decorator: options.delete(:item_decorator)}
-        options[:getter] ||= ->(*){ PagedCollection.new(self.send(name).page(1), nil, opts.merge({parent: self})) }
+        opts = {
+          no_curies: true,
+          item_class: options.delete(:item_class),
+          url: options.delete(:url),
+          item_decorator: options.delete(:item_decorator),
+          per: options.delete(:per) || Kaminari.config.default_per_page
+        }
+        options[:getter] ||= ->(*) do
+          # set # per page based on default, option value integer, or special :all
+          per = opts.delete(:per)
+          per = self.send(name).count if per == :all
```

Comment:
should these be calculated once outside the lambda? Or is it benefitial to lazily run the count?